packages/api/src/mappings/google/activity.ts

```diff
@@ -99,22 +208,48 @@ export const mapToActivity = (
     });
   }
 
-  if (googleSessions) {
-    const activitySessions = googleSessions.session.filter(
-      session => session.activityType !== sessionSleepType
-    );
-
-    const activityLogs = activitySessions.map(session => {
-      return {
-        metadata,
-        name: GoogleActivityTypes[session.activityType],
-        start_time: dayjs(Number(session.startTimeMillis)).toISOString(),
-        end_time: dayjs(Number(session.endTimeMillis)).toISOString(),
-      };
-    });
+  return activity;
+};
 
-    activity.activity_logs = activityLogs;
+function nanoTimeString(startTime: string | undefined): number {
+  if (startTime) {
+    const dateObject = new Date(startTime);
+    const timeInNanoseconds = dateObject.getTime() * 1e6;
+    return timeInNanoseconds;
   }
+  return 0;
+}
 
-  return activity;
-};
+function matchActivityTime(
+  activityStartTime: string | undefined,
+  activityEndTime: string | undefined,
+  startTimeNanos: number
+): boolean {
+  return (
+    nanoTimeString(activityStartTime) <= startTimeNanos &&
+    startTimeNanos < nanoTimeString(activityEndTime)
+  );
+}
+
+function formatNumber(num: number): number {
+  return parseInt((num * 100).toFixed(2)) / 100;
+}
```

Comment:
let's at least add a comment saying what this does to a number, too much effort to understand at a glance for future use